src/Npgsql/Internal/NpgsqlConnector.cs

```diff
@@ -451,6 +432,10 @@ internal async Task Open(NpgsqlTimeout timeout, bool async, CancellationToken ca
         {
             Debug.Assert(State == ConnectorState.Closed);
 
+            var canRetry = SslMode == SslMode.Prefer;
+
+        start:
```

Comment:
It may be better to simply have a local OpenCore method which accepts the SslMode, and calls itself recursively with SslMode.Disable. Aside from possibly simplifying the flow, this would also allow keeping resources readonly (e.g. _keepAliveTimer, CommandsInFlightReader, CommandsInFlightWriter...), since they would be initialized outside that local method.